src/inet_address.rs

```diff
@@ -38,3 +42,63 @@ impl InetAddress {
         }
     }
 }
+
+pub trait InetAddressExtManual {
+    fn to_bytes<'a>(&'a self) -> Option<InetAddressBytes<'a>>;
+}
+
+impl<O: IsA<InetAddress>> InetAddressExtManual for O {
+    /// Returns `None` in case the address has a native size different than 4 and 16.
+    fn to_bytes<'a>(&'a self) -> Option<InetAddressBytes<'a>> {
+        let size = self.get_native_size();
+        unsafe {
+            let bytes = gio_sys::g_inet_address_to_bytes(self.as_ref().to_glib_none().0);
+            if size == 4 {
+                // The `try_into` is used to convert from `&[u8]` to `&[u8; 4]`.
+                Some(InetAddressBytes::V4(
+                    from_raw_parts(bytes, 4)
+                        .try_into()
+                        .expect("failed to convert to array"),
+                ))
+            } else if size == 16 {
+                // The `try_into` is used to convert from `&[u8]` to `&[u8; 16]`.
+                Some(InetAddressBytes::V6(
+                    from_raw_parts(bytes, 16)
+                        .try_into()
+                        .expect("failed to convert to array"),
+                ))
+            } else {
+                None
+            }
+        }
+    }
+}
+
+impl From<IpAddr> for InetAddress {
+    fn from(addr: IpAddr) -> Self {
+        match addr {
+            IpAddr::V4(v4) => InetAddress::from_bytes(InetAddressBytes::V4(&v4.octets())),
+            IpAddr::V6(v6) => InetAddress::from_bytes(InetAddressBytes::V6(&v6.octets())),
+        }
+    }
+}
+
+impl Into<IpAddr> for InetAddress {
+    fn into(self) -> IpAddr {
+        let size = self.get_native_size();
+        unsafe {
+            let bytes = gio_sys::g_inet_address_to_bytes(self.to_glib_none().0);
+            if size == 4 {
+                let bytes = from_raw_parts(bytes, 4);
+                IpAddr::V4(Ipv4Addr::new(bytes[0], bytes[1], bytes[2], bytes[3]))
```

Comment:
Better via https://doc.rust-lang.org/std/net/struct.Ipv4Addr.html#impl-From%3C%5Bu8%3B%204%5D%3E

```suggestion
                IpAddr::V4(Ipv4Addr::from(bytes))
```